repo/mk/core.misc.channels.promote.mk

```diff
@@ -42,10 +38,30 @@ perform-promotion/%:
 		$(GIT_REMOTE)/$(PROMOTE_BRANCH)..$(TAG_COMMIT) | \
 		$(GREP) --color -i -E "break" || true
 	$(ECHO)
-	$(ECHO) "[Q   ] Still want to promote $(PROMOTE_BRANCH) to $(TAG)?"
+	$(ECHO) "[Q   ] Still want to promote $(TAG) to $(PROMOTE_BRANCH)?"
 	$(ECHO) "       Press ENTER to Continue."
 	$(ECHO) "       Press Ctrl+C to Cancel."
 	read -p ""
 	$(GIT) push --no-verify --force-with-lease $(GIT_REMOTE) \
 		$(TAG_COMMIT):refs/heads/$(PROMOTE_BRANCH)
 	$(GIT) fetch
+
+
+.PHONY: view-channels
+view-channels: deps-channels ## View the status of release channels
+	$(MAKE) -C $(SF_CHANNELS_DIR) show-channel-status
+
+
+.PHONY: show-channel-status
+show-channel-status:
+	$(ECHO)
+	$(ECHO_INFO) "Current channel status:";
+	$(ECHO)
+	# Because piping git log outputs an exit code 141 it needs to be handled
+	$(GIT) --no-pager log \
+		--oneline \
+		--graph \
+		--decorate \
+		--all | \
+		$(HEAD) -n 20 | \
```

Comment:
1. the exit code 141 comes from a consumer sending SIGPIPE to `git`, see https://www.ingeniousmalarkey.com/2016/07/git-log-exit-code-141.html
2. in your case, it's the `head` consumer, that will terminate input after it consumes 20 lines
3. to fix that, but also slightly more performant, is to pass `-20` to `git log` and remove the piping to `head` and `|| true` (which can hide other errors)

PS: I don't follow the piping to `xargs -0 echo` ❓

Reply:
Ah very nice! Didnt know I could pass `-20` to git log, I'll change that 👍 

Removing the xargs too, I thought I needed it. I probablyleft it in from a previous attempt that I never bothered checking if I still need